### ISSUE TITLE
Android support

### ISSUE THREAD

**Issue by abou7mied** [cdimascio/dotenv-kotlin#1]
Can this lib work on Android? here is the error I got when trying to use it
```
 java.lang.NoClassDefFoundError: Failed resolution of: Ljava/nio/file/Paths;
                                                             at io.github.cdimascio.dotenv.internal.DotenvReader.read(DotenvReader.kt:20)
                                                             at io.github.cdimascio.dotenv.internal.DotenvParser.parse(DotenvParser.kt:18)
                                                             at io.github.cdimascio.dotenv.DotenvBuilder.load(Dotenv.kt:43)
                                                             at io.github.cdimascio.dotenv.Dotenv$Instance.load(Dotenv.kt:9)
                                                             at io.github.cdimascio.dotenv.Dotenv.load(Dotenv.kt)
```

**Comment by cdimascio** [comment]
**tl/dr** short answer is it can work with a small change to the `java-dotenv` lib. please review the note below and let me know if this proposed change will satisfy your use case. if so, i'll deliver a new version  with this feature today.

---

@abou7mied i'm not an android expert, but i did do a few tests and found that android is not picking up  dot files from `assets` or `res/raw` directories.

in my local environment, i modified the `java-dotenv` library to allow callers to modify the default filename i.e. `.env`

Hence, I'm proposing to expose the `filename` property on dotenv, so that you can use a non-dot name e.g. `env`

by providing this functionality, i'm able to use `java-dotenv` with android 

**Here's how it would work**
1. Ensure you have an Android `assets` folder and create a file to hold you environment vars e.g. `env` (since, based on my testing dot files don't appear to work with android)
<p align="center">
<img src="https://user-images.githubusercontent.com/4706618/38046998-160bd8aa-328f-11e8-9db1-13020287e82b.png" width="300">
</p>

2. Use the newly proposed `filename` property to override the name `.env`
    ```
    val dotenv = dotenv {
        directory = "/assets"
        filename = "env" // tell dotenv to use the filename 'env', instead of '.env'
    }
    ```

I tested this locally and it works.
@abou7mied Would this satisfy your use case? If so I will release this functionality in a new version 
 today

**Comment by cdimascio** [comment]
@abou7mied I implemented the above proposal and delivered. look forward to your comments
you should be able to try it out using version 3.1.0

android details here:
https://github.com/cdimascio/java-dotenv#android-usage

**Comment by cdimascio** [comment]
Closing this out. Solution delivered in 1.3.0. @abou7mied would love to have your feedback. Thanks for the issue.
